src/System.Dynamic.Runtime/tests/Dynamic.Context/Conformance.dynamic.context.method.genmethod.genclass.cs

```diff
@@ -239,1557 +189,484 @@ public U Method_ReturnsU()
         {
             return new U();
         }
     }
 
-    public class MemberClassWithStructConstraint<T>
-        where T : struct
+    public class GenericClassWithStructConstraint<T> where T : struct
     {
-        public U Method_ReturnsU<U>()
-        {
-            return default(U);
-        }
+        public U Method_ReturnsU<U>() => default(U);
     }
 
-    public class MemberClassWithInterfaceConstraint<T>
-        where T : I
+    public class GenericClassWithInterfaceConstraint<T> where T : EmptyInterface
     {
         public dynamic Method_ReturnsDynamic<U, V>(int x, U u, V v) where V : U where U : T
         {
             return default(T);
         }
     }
-    #endregion
-}
-
 
-
-namespace ManagedTests.DynamicCSharp.Conformance.dynamic.context.method.genmethod.genclass.genclass001.genclass001
-{
-    using ManagedTests.DynamicCSharp.Conformance.dynamic.context.method.genmethod.genclass.genclassgenmeth.genclassgenmeth;
-    using ManagedTests.DynamicCSharp.Conformance.dynamic.context.method.genmethod.genclass.genclass001.genclass001;
-    // <Title> Tests generic class generic method used in static method body.</Title>
-    // <Description>
-    // </Description>
-    // <RelatedBugs></RelatedBugs>
-    //<Expects Status=success></Expects>
-    // <Code>
-    using System;
-
-    public class Test
+    public class GenericMethodGenericClassTests
     {
+        public class NestedClass : GenericMethodGenericClassTests { }
+
         [Fact]
-        public static void DynamicCSharpRunTest()
+        public static void StaticMethodBody1()
         {
-            Assert.Equal(0, MainMethod());
+            dynamic d = new GenericClass<int>();
+            int ret = d.Method_ReturnsT<string>(string.Empty, "ab", null);
+            dynamic d1 = string.Empty;
+            dynamic d2 = "ab";
+            ret += d.Method_ReturnsT(d1, d2, null);
+            Assert.Equal(0, ret);
         }
 
-        public static int MainMethod()
+        [Fact]
+        public static void IfElseBlock()
         {
-            if (Test.TestMethod() != 0)
-                return 1;
-            return 0;
+#if c1
+dynamic mc = new GenericClass<int>();
+return (Test)mc.Method_ReturnsUConstraint<Test>(new Test());
+#else
+            dynamic d = new GenericClass<int>();
+            Assert.Null(d.Method_ReturnsUConstraint<string>("1"));
+            Assert.Throws<RuntimeBinderException>(() => (int)d.Method_ReturnsUConstraint<string>(1));
+#endif
         }
 
-        public static int TestMethod()
+        [Fact]
+        public static void ImplicitlyTypedVariableInitializer()
         {
-            dynamic mc = new MemberClass<int>();
-            int ret = mc.Method_ReturnsT<string>(string.Empty, "ab", null);
-            dynamic d1 = string.Empty;
-            dynamic d2 = "ab";
-            ret += mc.Method_ReturnsT(d1, d2, null);
-            return ret;
+            dynamic d = new GenericClass<string>();
```

Comment:
Is this supposed to be equivalent of ManagedTests.DynamicCSharp.Conformance.dynamic.context.method.genmethod.genclass.genclass003.genclass003.MainMethod ?